src/controllers/authController.ts

```diff
@@ -0,0 +1,22 @@
+import { Request, Response } from "express";
+import { constructAuthUrl, contructLoginUrl } from "../services/authService";
+
+/**
+ * Initiates OAuth login flow by redirecting to the WCA authorization endpoint
+ */
+export const login = async (req: Request, res: Response): Promise<void> => {
+  // Redirect user to authorization page
+  const authUrl = constructAuthUrl();
+  res.redirect(authUrl);
+};
+
+/**
+ * Handles OAuth callback, exchanges authorization code for access token,
+ * and fetches user information
+ */
+export const callback = async (req: Request, res: Response): Promise<void> => {
+  const { code } = req.query;
```

Comment:
How about `const code = req.query.code` (since you are extracting only one variable)?